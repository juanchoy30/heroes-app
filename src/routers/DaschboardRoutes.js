import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { Navbar } from '../components/ui/Navbar'
import { DcScreen } from '../components/dc/DcScreen'
import { HeroScreen } from '../components/heroes/HeroScreen'
import { MarvelScreen } from '../components/marvel/MarvelScreen'

export const DaschboardRoutes = () => {
    return (
        <>
            <Navbar />
            <div className="container mt-2">
                <Switch>
                    <Route exact path="/marvel" component={ MarvelScreen } />
                    <Route exact path="/hero/:heroId" component={ HeroScreen } />
                    <Route exact path="/dc/" component={ DcScreen } />
                    <Redirect to="/marvel" />
                </Switch>
            </div>

        </>
    )
}
