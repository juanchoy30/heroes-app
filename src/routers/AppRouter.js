import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route
  } from 'react-router-dom';
import { LoginScreen } from '../components/login/LoginScreen';
import { DaschboardRoutes } from './DaschboardRoutes';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Switch>
                    <Route exact path="/login" component={ LoginScreen } />
                    <Route path="/" component={ DaschboardRoutes } />
                </Switch>
            </div>
        </Router>
    )
}
