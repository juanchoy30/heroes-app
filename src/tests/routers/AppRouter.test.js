import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'; // *1
import { configure, mount } from 'enzyme'; // *1
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';

configure({adapter: new Adapter()}); 

// *1 This feature was not in the course. I found it on stackOverFlow, It seems to be
// a problem with the wojtekmaj adapter patch for enzima-react17. 

describe('<AppRouter /> Tests', () => {

    const contexValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    };
    
    test('should render loggin component if not authenticated', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={ contexValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();

    });

    test('should render Marvel component if authenticated', () => {

        const contexValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Juan'
            }
        };
        
        const wrapper = mount(
            <AuthContext.Provider value={ contexValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper.find('.navbar').exists() ).toBe( true );

    });
    
    

})
