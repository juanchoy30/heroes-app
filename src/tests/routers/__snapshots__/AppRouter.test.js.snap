// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> Tests should render loggin component if not authenticated 1`] = `ReactWrapper {}`;
